src/security/builtin_plugins/cryptographic/src/crypto_objects.c

```diff
@@ -0,0 +1,593 @@
+/*
+ * Copyright(c) 2006 to 2019 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <assert.h>
+#include <string.h>
+#include "dds/ddsrt/heap.h"
+#include "dds/ddsrt/hopscotch.h"
+#include "dds/ddsrt/types.h"
+#include "crypto_objects.h"
+#include "crypto_utils.h"
+
+bool crypto_object_valid(CryptoObject *obj, CryptoObjectKind_t kind)
+{
+  return (obj && obj->kind == kind && obj->handle == (int64_t)(uintptr_t)obj);
+}
+
+static uint32_t crypto_object_hash(const void *obj)
+{
+  const CryptoObject *object = obj;
+  const uint64_t c = 0xF6D6E408505;
```

Comment:
I'm curious why you switched to such a small constant. With this constant the LSB of the handle affects approximately roughly the lowest 12 bits of the computed hash. The hash table implementation then takes the low order bits of the calculated hash, and so (unless I'm mistaken) for tables larger than ~2k buckets the hashing is really insufficient. (See also https://en.wikipedia.org/wiki/Universal_hashing)

Reply:
Thanks for catching this, I missed a parantheses in the macro definition and so I calculated the wrong value (I should have read some background info on hashing, then I'd notice this myself). I fixed this in my latest commit.